src/FSharp.Control.TaskSeq/TaskSeqInternal.fs

```diff
@@ -908,13 +908,11 @@ module internal TaskSeqInternal =
                 go <- step
         }
 
-        interface IAsyncDisposable with
-            override _.DisposeAsync() =
-                if not (isNull _rwLock) then
+        interface IDisposable with
+            override _.Dispose() =
+                if isNull _rwLock |> not then
```

Comment:
I originally wrote something that: this is a behavioral change and backwards incompatible [...]

However, the context wasn't clear while reviewing (I thought it applied to an exposed type), so I looked deeper and this is clearly only used internally. As such, it matters little, and may even be a (small) perf improvement.

But, I would actually like to have a test that checks that the `ReaderWriterLock` is properly released. And in fact, we should test this implementation a bit further, probably. Not part of this PR, that's for another time.

Reply:
I'll let you resolve this if you want to make a note first

Reply:
I'll lift this one into a separate PR, that way these functional changes won't be hidden in a refactoring branch.

See: #227